abel/tools/circularize.py

```diff
@@ -0,0 +1,291 @@
+# -*- coding: utf-8 -*-
+from __future__ import division
+import numpy as np
+from scipy import ndimage
+from scipy.interpolate import UnivariateSpline
+from scipy.optimize import leastsq
+
+import abel
+
+#########################################################################
+# circularize.py
+# 
+# Image circularization by following peak intensity vs angle
+# see https://github.com/PyAbel/PyAbel/issues/186 for discussion
+# and https://github.com/PyAbel/PyAbel/pull/195
+#
+# Steve Gibson and Dan Hickstein - ideas/code 
+# Jason Gascooke - ideas
+#
+# February 2017
+#########################################################################
+
+
+def circularize_image(IM, method="lsq", center=None, radial_range=None,
+                      dr=0.5, dt=0.5, smooth=0, ref_angle=None, 
+                      inverse=False, return_correction=False):
+    """
+    Corrects image distortion on the basis that the structure (Newton spheres)
+    should be circular.  
+
+    This function is especially useful for correcting the image obtained with
+    a velocity-map-imaging spectrometer with imperfect electro-static lenses,
+    allowing a high-resolution 1D photoelectron distribution to be extracted.
+
+    The algorithm splits the image into "slices" at many different angles
+    (set by `dt`) and compares the radial intensity profile of adjacent slices.
+    A scaling factor is found which aligns each slice profile with the previous
+    slice. The image is then corrected using a spline function that smoothly 
+    connects the discrete scaling factors as a continuous function of angle.
+
+    This circularization algorithm should only be applied to a well-centered
+    image.
+
+
+    Parameters
+    ----------
+    IM : numpy 2D array
+
+    method : str
+        method to align slice profiles,
+        "argmax" - compare intensity-profile.argmax() of each radial slice.
+        "lsq" - determine radial scaling factor by minimizing the
+                intensity-prfoile with an adjacent slice intensity-profile.
+
+    center : str, float tuple, or None
+        pre-center image using :func:`abel.tools.center.center_image`
+        "com", "convolution", "gaussian", "image_center", "slice", or
+        float tuple center (y, x).
+
+    radial_range : tuple, or None
+        limit slice comparison to the radial range tuple (rmin, rmax), in
+        pixels, from the image center. Use to determine the distortion 
+        correction associated with particular peaks. 
+
+    dr : float
+        Radial grid size for the polar coordinate image, default = 0.5 pixel.
+        This is passed to :func:`abel.tools.polar.reproject_image_into_polar`.
+
+        Small values may improve the distortion correction, which is often of
+        sub-pixel dimensions, at the cost of reduced signal to noise for the
+        slice intensity profile. 
```

Comment:
"As a general rule, dr should be significantly smaller than the "feature size" in the image."